### PR TITLE
User settings

### PR DESCRIPTION
**Pull request by eddovandenboom**
Implemented user settings: useSolver (saves whether useSolver is selected), pbnFilePath (saves location of current pbn file) and boardNumber (saves current board number). When the application starts, the saved settings are loaded.

### CODE REVIEW

File: Tosr/Form1.cs [hunk 1]
```diff
@@ -54,8 +55,22 @@ private async void Form1LoadAsync(object sender, EventArgs e)
...
-            StartBidding();
+
+            // Load user settings
+            toolStripMenuItemUseSolver.Checked = Settings.Default.useSolver;
+            numericUpDown1.Value = Settings.Default.numberOfHandsToBid;
+            if (File.Exists(Settings.Default.pbnFilePath))
+            {
+                pbn.Load(Settings.Default.pbnFilePath);
+                pbnFilename = Path.GetFileName(Settings.Default.pbnFilePath);
+                boardNumber = Settings.Default.boardNumber;
```
Comment (YkeSmit1) [inline Tosr/Form1.cs:66]: There is a chance boardnumber is larger then the highest number in the Pnb file. For example, if it is overwritten by a third party application. So please check for this.

File: Tosr/Form1.cs [hunk 2]
```diff
@@ -224,6 +239,7 @@ private async void ButtonBatchBiddingClickAsync(object sender, EventArgs e)
                     var progress = new Progress<int>(report => toolStripStatusLabel1.Text = $"Hands done: {report}");
                     pbn = batchBidding.Execute(pbn.Boards.Select(x => x.Deal), progress, cancelBatchbidding.Token);
                 });
+                Settings.Default.pbnFilePath = "";
```
Comment (YkeSmit1) [inline Tosr/Form1.cs:242]: I think we should also save the pbnFilePath in the form close. To make it consistent. There should be only one place to store all the settings. To achieve this, you need to change the variable PbnFilename to PbnFilepath. And only save it if the file exists. That would make the most sense to the user.

**Review by YkeSmit1 — CHANGES_REQUESTED** [review]

File: Tosr/Form1.cs (hunk 1 above)
Comment (eddovandenboom) [reply Tosr/Form1.cs:66]: I check this now in the method LoadCurrentBoard: if the board number doesn't exist, then I set it equal to 1

Comment (eddovandenboom) [reply Tosr/Form1.cs:66]: I check this now in the method LoadCurrentBoard: if the board number doesn't exist, then I set it equal to 1

**Review by YkeSmit1 — APPROVED** [review]

**Pull request merged**